src/org/zaproxy/zap/extension/ascanrulesBeta/BufferOverflow.java

```diff
@@ -0,0 +1,180 @@
+	private static Logger log = Logger.getLogger(BufferOverflow.class);
+	
+	@Override
+	public int getId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+	
+	public String getOther() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "other");
+	}
+	
+	@Override
+	public void init() {
+
+	}
+
+	/*
+	 * This method is called by the active scanner for each GET and POST parameter for every page 
+	 * @see org.parosproxy.paros.core.scanner.AbstractAppParamPlugin#scan(org.parosproxy.paros.network.HttpMessage, java.lang.String, java.lang.String)
+	 */
+	@Override
+	public void scan(HttpMessage msg, String param, String value) {
+	
+		if (this.isStop()) { // Check if the user stopped things
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+this.getName()+" Stopping.");
+			}
+			return; // Stop!
+		}
+		
+		try {
+			// This is where you change the 'good' request to attack the application
+			// You can make multiple requests if needed
+			String checkStringHeader1 = "Connection: close";  // Un natural close
+			// Always use getNewMsg() for each new request
+			msg = getNewMsg();
+			String returnAttack = randomCharacterString(2100);
+			setParameter(msg, param, returnAttack);
+			sendAndReceive(msg);
+			HttpResponseHeader requestReturn = msg.getResponseHeader();			
+			// This is where BASE baseResponseBody was you detect potential vulnerabilities in the response
+    		String chkerrorheader = requestReturn.getHeadersAsString();
+    		log.debug("Header: "+ chkerrorheader);
+    		if (msg.getResponseHeader().getStatusCode() == HttpStatusCode.INTERNAL_SERVER_ERROR && chkerrorheader.contains(checkStringHeader1))
+    		{
+    			log.debug("Found Header");
+    			bingo(getRisk(), 
+    					Alert.CONFIDENCE_MEDIUM, 
+    					this.getBaseMsg().getRequestHeader().getURI().toString(), 
+    					param, 
+    					msg.getRequestHeader().toString(), 
+    					this.getOther() ,
+    					msg);
+    			return;
+    		}
+    			
+				return;	
+
+			
+		} catch (Exception e) {
+			log.error(e.getMessage(), e);
+		}	
+	}
+
+	@Override
+	public int getRisk() {
+		return Alert.RISK_MEDIUM;
+	}
+
+	@Override
+	public int getCweId() {
+		// The CWE id
+		return 120;
+	}
+
+	@Override
+	public int getWascId() {
+		// The WASC ID
+		return 7;
+	}
+	
+	private String randomCharacterString(int length)
+	{
+		StringBuilder sb1 = new StringBuilder();

```

Comment:
The initial capacity of the StringBuilder should be the parameter _length_ (to avoid unnecessary re-allocations of internal array).